### ISSUE TITLE
Bug: [strict-boolean-expressions] false positive for string | object | undefined with allowNullableString enabled

### ISSUE THREAD

**Issue by hgl** [typescript-eslint/typescript-eslint#12045]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.9.2&fileType=.tsx&code=C4TwDgpgBAYg9nKBeKBnYAnAlgOwOZQA%2BUA3lDgIYC2EAXGprgQL5FQCuOAJhAGa4QuAKCEAbCMCi8E9eHBFZeUABTS4ASlJCoUAIxDmQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZfKsugIwHs%2BSAIZN6AD2LQUySnyaoMAbUiJo0PtEgAaTJCHx4fAO4A5BPCE8kAZU7MA5ujCdYiAL4BdcGDcg3QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type Foo = string | { name: string } | undefined

let foo: Foo

if (foo) {
  1
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/strict-boolean-expressions": ["error", {"allowNullableString": true}]
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Not error.

### Actual Result

`if (foo) {` has the error:

```
Unexpected value in conditional. A boolean expression is required. 5:5 - 5:8
```

**Comment by hgl** [comment]
Maybe it's because typescript-eslint doesn't support a combination of enabled nullables? It would be great if it could.

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/4969
